brave/src/main/java/brave/propagation/CurrentTraceContext.java

```diff
@@ -45,4 +48,59 @@
       return () -> local.set(previous);
     }
   }
-}
\ No newline at end of file
+
```

Comment:
note implementations can completely replace any of these features below as they are non-final